web-app-router/scenes/Portal/Teams/TeamId/Apps/AppId/Profile/Gallery/page/index.tsx

```diff
@@ -37,8 +38,9 @@ export const AppProfileGalleryPage = ({ params }: AppProfileGalleryProps) => {
 
   const app = data?.app[0];
 
-  if (!app) {
-    <Error statusCode={404} title="Action not found" />;
+  if (loading) return <div></div>;
+  else if (!app) {
```

Comment:
@andy-t-wang You can also get `error` from `useFetchAppMetadataQuery` to add to this condition.

Reply:
let's add it in other places too, in StoreInfo, for example